sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -1058,7 +1066,8 @@ void Scheduler::GraphBuilder::cleanupFinishedCommands(
     // Update dependency users
     for (DepDesc &Dep : Cmd->MDeps) {
       Command *DepCmd = Dep.MDepCommand;
-      DepCmd->MUsers.erase(Cmd);
+      if (DepCmd)
+        DepCmd->MUsers.erase(Cmd);
```

Comment:
Not a fan of introducing dependencies on `nullptr` and having to handle them here. I assume this is a side effect of `EmptyCommand::addRequirement` on L358. Can this dependency be avoided?